src/scf/newton.jl

```diff
@@ -0,0 +1,360 @@
+    _, H = energy_hamiltonian(basis, ψ, occupation; ρ=ρ)
+
+    [proj_tangent_kpt(H.blocks[ik]*ψ[ik], ψ[ik]) for ik = 1:Nk]
+end
+
+"""
+    proj_tangent_kpt(δψk, ψk)
+
+Computation the projection of δψk onto the tangent space defined at ψk, where
+δψk and ψk are defined at the same kpoint.
+"""
+proj_tangent_kpt(δψk, ψk) = δψk - ψk * (ψk'δψk)
+function proj_tangent_kpt!(δψk, ψk)
+    mul!(δψk, ψk, ψk'δψk, -1, 1)
+end
+"""
+    proj_tangent(δψ, ψ)
+
+Computation the projection of δψ onto the tangent space defined at ψ.
+"""
+function proj_tangent(δψ, ψ)
+    [proj_tangent_kpt(δψ[ik], ψ[ik]) for ik = 1:size(ψ,1)]
+end
+function proj_tangent!(δψ, ψ)
+    [proj_tangent_kpt!(δψ[ik], ψ[ik]) for ik = 1:size(ψ,1)]
+end
+
+# Operators
+
+"""
+    apply_Ω(basis::PlaneWaveBasis, δψ, ψ, H)
+
+Apply Ω to an element in the tangent space to ψ.
+"""
+function apply_Ω(basis::PlaneWaveBasis, δψ, ψ, H)
+    Nk = length(basis.kpoints)
+
+    Hψ = [H.blocks[ik] * ψ[ik] for ik = 1:Nk]
+    ψHψ = [(ψ[ik]*Hψ[ik]' + Hψ[ik]*ψ[ik]') / 2 for ik = 1:Nk]
+    λ = [ψ[ik]'Hψ[ik] for ik = 1:Nk]
+
+    δψ = proj_tangent(δψ, ψ)
+    Ωδψ = [H.blocks[ik] * δψ[ik] - ψHψ[ik] * δψ[ik] for ik = 1:Nk]
+    Ωδψ = Ωδψ - [δψ[ik]*λ[ik] for ik = 1:Nk]
+    proj_tangent!(Ωδψ, ψ)
+end
+
+"""
+    compute_dρ(ψ, δψ)
+
+Compute δρ = Σi ψi*conj(δψi) + conj(ψi)*δψi
+"""
+@views function compute_dρ(basis::PlaneWaveBasis, ψ, δψ, ρ, occupation)
+                           n_spin = basis.model.n_spin_components
+
+    δρ_fourier = zeros(complex(eltype(ρ)), size(ρ)...)
+    for (ik, kpt) in enumerate(basis.kpoints)
+        δρk = zeros(eltype(ρ), basis.fft_size)
+        for i = 1:size(ψ[ik], 2)
+            ψki_real = G_to_r(basis, kpt, ψ[ik][:, i])
+            δψki_real = G_to_r(basis, kpt, δψ[ik][:, i])
+            δρk .+= occupation[ik][i] .* (conj.(ψki_real) .* δψki_real +
+                                          conj.(δψki_real) .* ψki_real)
+        end
+        # Check sanity in the density, we should have ∫δρ = 0
+        T = real(eltype(δρk))
+        if abs(sum(δρk)) > sqrt(eps(T))
+            @warn("Mismatch in δρ", sum_δρ=sum(δρk))
+        end
+        δρk_fourier = r_to_G(basis, complex(δρk))
+        lowpass_for_symmetry!(δρk_fourier, basis)
+        accumulate_over_symmetries!(δρ_fourier[:, :, :, kpt.spin], δρk_fourier,
+                                    basis, basis.ksymops[ik])
+    end
+    mpi_sum!(δρ_fourier, basis.comm_kpts)
+    count = sum(length(basis.ksymops[ik]) for ik in 1:length(basis.kpoints)) ÷ n_spin
+    count = mpi_sum(count, basis.comm_kpts)
+    G_to_r(basis, δρ_fourier) ./ count
+end
+
+"""
+    apply_K(basis::PlaneWaveBasis, δψ, ψ, ρ, occupation)
+
+Compute the application of K to an element in the space tangent to ψ.
+"""
+function apply_K(basis::PlaneWaveBasis, δψ, ψ, ρ, occupation)
+    Nk = length(basis.kpoints)
+
+    δψ = proj_tangent(δψ, ψ)
+    dρ = compute_dρ(basis, ψ, δψ, ρ, occupation)
+    dV = apply_kernel(basis, dρ; ρ=ρ)
```

Comment:
we use sometimes delta, sometimes d; would be good to settle on something and harmonize

Reply:
Seems we mostly use deltax instead of dx, so we should do it consistently (including in chi0, dielectric, etc). Mind making a pass over all the code and fix it? Or just open an issue

Reply:
Okay, I'll try to do that.

Reply:
Not needed in chi0, I just changed some stuff in dielectric. Other files where it should be done ?

Reply:
grep for dV, drho, dpsi, docc, depsF and look in mixing.jl. Otherwise we're probably good!

Reply:
Okay, should be good. I included it in the PR, but in a specific commit.